shared/src/utils/validators/is-string-literal-union.ts

```diff
@@ -0,0 +1,26 @@
+import type { ValidationOptions, ValidationArguments } from 'class-validator';
+import { StrictObject } from '../strict-object';
+import { makeValidator } from './make-validator';
+
+export function isStringLiteralUnion<T extends string>(
+    allowedValues: { [key in T]: true },
+    valueToBeValidated: unknown
+): boolean {
+    return (
+        typeof valueToBeValidated === 'string' &&
+        StrictObject.keys(allowedValues).includes(valueToBeValidated as T)
+    );
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function IsStringLiteralUnion<T extends string>(
+    allowedValues: { [key in T]: true },
+    validationOptions?: ValidationOptions
+) {
+    return makeValidator(
+        'isStringLiteralUnion',
+        (value: unknown, args?: ValidationArguments) =>
+            isStringLiteralUnion<T>(allowedValues, value),
```

Comment:
Do we need the second function?

```suggestion
        (value: unknown, args?: ValidationArguments) =>
            allowedValues[value as T],
```

Reply:
This is a convention for `class-validator` validators: They all have an `@Is...` decorator, to be used for, well, decorating properties, and a `is...` function, used when a manual validation is required (we use this e.g. with `isUUID` in this PR.